File: core/modules/mock-modules.js
module.exports = {
  'game-understanding': {
    async execute(action, context) {
      return {
        blueprint: {
          genre: 'roguelike',
          artStyle: 'unknown',
          source: context.input || 'image'
        }
      };
    }
  },

  'blueprint-generator': {
    async execute(action, context) {
      return {
        tasks: [
          'generate_player',
          'generate_environment',
          'generate_ui'
        ]
      };
    }
  },

  'export-package': {
    async execute(action, context) {
      return {
        package: {
          path: 'generated-project/',
          ready: true
        }
      };
    }
  }
};

File: core/runtime/mock_modules.py
from .module_registry import BaseModule


class GameUnderstandingModule(BaseModule):
    id = "game-understanding"

    def execute(self, action, context):
        return {
            "blueprint": {
                "genre": "unknown",
                "artStyle": "analyzed-style",
                "coreLoop": ["play", "progress"]
            }
        }


class BlueprintGeneratorModule(BaseModule):
    id = "blueprint-generator"

    def execute(self, action, context):
        return {
            "assetTasks": [
                "character",
                "tileset",
                "ui"
            ]
        }


class AssetPackageModule(BaseModule):
    id = "asset-package"

    def execute(self, action, context):
        return {
            "package": "generated-project/assets"
        }
